### PR TITLE
Ceph kubeflow

### PR DESCRIPTION
**Pull request by yangatgithub**
Fix Ceph provisioning issues in AWS for StorageClass creation, which is required by Kubeflow.

### CODE REVIEW

**Comment by supertetelman** [comment]
It looks like you accidentally pulled in a bunch of the Docker changes from Adam's other branch here. Can we pull those out to avoid conflicts?

File: scripts/k8s_deploy_rook.sh [hunk 1]
```diff
@@ -6,7 +6,9 @@
 # `helm upgrade --namespace rook-ceph rook-ceph rook-release/rook-ceph --version v0.9.0-174.g3b14e51`
 
 HELM_ROOK_CHART_REPO="${HELM_ROOK_CHART_REPO:-https://charts.rook.io/release}"
-HELM_ROOK_CHART_VERSION="${HELM_ROOK_CHART_VERSION:-v1.1.1}"
+
+# Always use the latest unless any specific reason for a particular release
+#HELM_ROOK_CHART_VERSION="${HELM_ROOK_CHART_VERSION:-v1.1.1}"
```
Comment (supertetelman) [inline scripts/k8s_deploy_rook.sh:11]: This is bad practice, we want to always peg to a specific version.

Can you update this with whatever the current version is?

File: scripts/k8s_deploy_rook.sh [hunk 2]
```diff
@@ -16,7 +18,7 @@ helm repo add rook-release "${HELM_ROOK_CHART_REPO}"
 # We need to dynamically set up Helm args, so let's use an array
 helm_install_args=("--namespace" "rook-ceph"
                    "--name" "rook-ceph"
-		   "--version" "${HELM_ROOK_CHART_VERSION}"
+#		   "--version" "${HELM_ROOK_CHART_VERSION}"
```
Comment (supertetelman) [inline scripts/k8s_deploy_rook.sh:21]: Same here, we should remove this and just use the current version for now.

File: services/rook-cluster.yml [hunk 3]
```diff
@@ -16,7 +16,7 @@ spec:
     enabled: true
   storage:
     useAllNodes: true
-    useAllDevices: false
+    useAllDevices: true
```
Comment (supertetelman) [inline services/rook-cluster.yml:19]: What does this do? Would this be an issue for larger clusters?

File: playbooks/k8s-cluster.yml [hunk 4]
```diff
@@ -49,6 +49,17 @@
   tags:
     - bootstrap
 
+- hosts: all
```
Comment (supertetelman) [inline playbooks/k8s-cluster.yml:52]: We need to pull this commit out.

File: config.example/group_vars/all.yml [hunk 5]
```diff
@@ -226,3 +226,25 @@ deepops_dir: /opt/deepops
 # Directory for python virtualenv
 # Roles: K8s GPU operator, GPU plugin, OpenShift/K8s
 deepops_venv: '{{ deepops_dir }}/venv'
+
```
Comment (supertetelman) [inline config.example/group_vars/all.yml:229]: We need to pull this commit out.

**Comment by yangatgithub** [comment]
Later on after additional testing I found this works without the “deviceFilter” statement I tried earlier, this approach is more flexible and should work in large cluster.  If users want more granular control, they probably can specifically define the “device” they’d like to use.

**Comment by yangatgithub** [comment]
Accidentally add PR519 branch, will take it out and re-commit with other changes

**Pull request closed without merging**